_episodes_rmd/03-r-data-analysis.Rmd

```diff
@@ -33,258 +32,315 @@ knitr_fig_path("03-")
 
+> ## Pro tip: Counting
+> 
+> Often we want to count how many of each category we have in our dataset. 
+> To do this, we can use the function `n()` within `summarize()`. 
+> For instance, here we count the number of countries in each continent for which we have data in 2019:
+> 
+> ```{r}
+> smoking %>% 
+>   filter(year == 2019) %>% 
+>   group_by(continent) %>% 
+>   summarize(count = n())
+> ```
+> 
+{: .callout}
+
+
 > ## Grouping the data
 >
-> Try calculating the average life expectancy by continent.
+> Try calculating the average percent of smokers by continent.
 >
 > > ## Solution
 > >
 > > ```{r GroupByChallenge1}
-> > gapminder_data %>%
+> > smoking %>%
 > > group_by(continent) %>%
-> > summarize(average=mean(lifeExp))
+> > summarize(avg_smoke_pct=mean(smoke_pct))
 > > ```
 > > {: .source}
 > >
-> > By combining `group_by()` and `summarize()` we are able to calculate the mean life expectancy by continent.
+> > By combining `group_by()` and `summarize()` we are able to calculate the mean percent of smokers by continent.
 > {: .solution}
 {: .challenge}
 
-You can also create more than one new column when you call `summarize()`. To do so, you must separate your columns with a comma. Building on the code from the last exercise, let's add a new column that calculates the minimum life expectancy for each continent. 
 
-```{r GapLifeExpMinMax}
-gapminder_data %>%
-  group_by(continent) %>%
-  summarize(average=mean(lifeExp), min=min(lifeExp))
-```
+> ## Bonus: plotting review
+>
+> Make a line plot of year vs. average percent of smokers to be able to more easily visualize the relationship between these two variables. 
+>
+> > ## Solution
+> >
+> > ```{r}
+> > smoking %>%
+> >   group_by(year) %>%
+> >   summarize(average=mean(smoke_pct)) %>% 
+> >   ggplot(aes(x = year, y = average)) +
+> >   geom_line()
+> > ```
+> > {: .source}
+> >
+> > You can pipe your summarized data right to ggplot! 
+> {: .solution}
+{: .challenge}
 
 
 ## Make new variables with `mutate()` {#make-new-variables-with-mutate}
-
 [*Back to top*](#contents)
 
 Each time we ran `summarize()`, we got back fewer rows than passed in. We either got one row back, or one row per group. But sometimes we want to create a new column in our data without changing the number of rows. The function we use to create new columns is called `mutate()`.
 
-We have a column for the population and the GDP per capita. If we wanted to get the total GDP, we could multiply the per capita GDP values by the total population. Here's what such a `mutate()` command would look like:
+We have a column for the population and the percent of people who smoke. If we wanted to get an estimate of the number of people who smoke, we could multiply the percentages by the total population. Here's what such a `mutate()` command would look like:
 
 ```{r GapMutate}
-gapminder_data %>%
-  mutate(gdp = pop * gdpPercap)
+smoking %>%
+  mutate(smoke_n = pop * smoke_pct)
```

Comment:
Pretty sure we changed these to actual percentages, right? (aka multiplied by 100)
```suggestion
  mutate(smoke_n = pop * smoke_pct/100)
```